### PR TITLE
Bluetooth: controller: Remove BT_HCI_RAW dependency

### PR DESCRIPTION
**Pull request by thomasstenersen**
Removes the dependency on BT_HCI_RAW from BT_CTLR_ASSERT_HANDLER.

For example, when testing the full stack in CI it is useful to re-define the assert handler to get additional system information when an assert is triggered. In this case, BT_HCI_RAW is not necessarily set, since the application is usually running the full host+controller on a single chip.

### CODE REVIEW

**Review by cvinayak — CHANGES_REQUESTED** [review]
Removing the dependency will cause the controller asserts to use uninitialized UART causing conflicts with application's use of UART, if any. This depend is in place in order to have HCI error event be generated when building controller only (ex, hci_uart). That said....hmmm... hci_usb is broken! controller will use UART for error!

**Comment by thomasstenersen** [comment]
As discussed on Slack, the change in this PR does not affect UART in any way. It just allows the user the define its own assertion handler without needing to set BT_HCI_RAW when it's not used.

File: subsys/bluetooth/controller/Kconfig [hunk 1]
```diff
@@ -733,7 +733,6 @@ comment "BLE Controller debug configuration"
 
 config BT_CTLR_ASSERT_HANDLER
 	bool "Bluetooth Controller Assertion Handler"
```
Comment (cvinayak) [inline subsys/bluetooth/controller/Kconfig]: Rename to "Application Defined Assertion Handler"

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/Kconfig (hunk 1 above)
Comment (thomasstenersen) [reply subsys/bluetooth/controller/Kconfig]: Done.

**Review by jhedberg — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Pull request merged**